src/main/java/com/cburch/logisim/gui/log/OptionsPanel.java

```diff
@@ -479,7 +479,8 @@ String getText() {
       @Override
       public java.awt.Component getListCellRendererComponent(JList<?> list,
           Object w, int index, boolean isSelected, boolean cellHasFocus) {
-        String s = renderAsText((E)w);
+        @SuppressWarnings("unchecked")
+        final String s = renderAsText((E) w);
```

Comment:
@zdimension: Do you have a better idea how to fix this warning?

Reply:
`w` is an Object because the JComboBox is supposed to support objects of a different type than the generic parameter of the instance; when it is in editable mode for instance (then the current item can be a String).

I can see how to fix the warning at line 464, I think

    E getValue() {
      return (E)getSelectedItem();
    }

can be replaced by

    E getValue() {
      return getItemAt(getSelectedIndex());
    }

Which makes it type safe. But in your case, since the method is overridden, we can't just change the signature and expect it to work, so I don't think there's anything better than unsafely casting `w` to `E`. I'll have a look at this to see if there is any hidden Swing way to do it.